### PR TITLE
Fixes `Error.stack` is being `undefined` when `Error.stackTraceLimit==0`

### PR DESCRIPTION
**Pull request by obastemur**
Fixes https://github.com/Microsoft/ChakraCore/issues/589

### CODE REVIEW

**Comment by obastemur** [comment]
Ping.. This can also go into 1.2 ? Change is very simple actually. IMHO Git diff messed up a bit.

File: lib/Runtime/Language/JavascriptExceptionOperators.cpp [hunk 1]
```diff
@@ -682,50 +682,46 @@ namespace Js
         // NOTE Don't set the throwing exception here, because we might need to box it and will cause a nested stack walker
         // instead, return it to be set in WalkStackForExceptionContext
 
-        if (stackCrawlLimit == 0)
+        JavascriptExceptionContext::StackTrace *stackTrace = RecyclerNew(scriptContext.GetRecycler(), JavascriptExceptionContext::StackTrace, scriptContext.GetRecycler());
+        if (stackCrawlLimit > 0)
```
Comment (jianchun) [inline lib/Runtime/Language/JavascriptExceptionOperators.cpp]: You moved RecyclerNew(...StackTrace...) outside of BEGIN/END_TRANSLATE_EXCEPTION block. This changed OOM logic. This code is called during exception throw. If it OOMs here and throws a different exception, that would be undesirable.

Comment (obastemur) [inline lib/Runtime/Language/JavascriptExceptionOperators.cpp]: > You moved RecyclerNew(...StackTrace...) outside of BEGIN/END_TRANSLATE_EXCEPTION block. 

I had checked couple of usages of RecyclerNew and it had looked fine reaching outside of the block. 

I guess the intention is to make this part safe from any possible throw. @jianchun Thanks for the review. I've updated the PR and kept that in.

**Comment by jianchun** [comment]
Change LGTM. Thanks @obastemur

**Comment by jianchun** [comment]
@obastemur Apologize for delay. I was told this issue is not critical and fix should go to master (post rs1). Could you please rebase this fix to latest master?

In addition, could you please also update related unit tests in  `test\StackTrace`? There are existing tests on `Error.stackTraceLimit == 0`. You can consider in `LongCallStackThrow.js` change `Dump(TrimStackTracePath(e.stack) || e)` to just `Dump(TrimStackTracePath(e.stack))` and see what changes it brings. It looks the `|| e` part was to workaround this issue.

**Pull request closed without merging**